src/main/groovy/com/bugsnag/android/gradle/BugsnagUploadNdkTask.groovy

```diff
@@ -155,67 +153,21 @@ class BugsnagUploadNdkTask extends BugsnagMultiPartUploadTask {
     }
 
     /**
-     * Outputs the contents of the outReader from the objdump command to the outputFile
-     * Removes redundant address lines to minimize file size
+     * Outputs the contents of stdout into the gzip file output file
      *
-     * @param outReader The objdump output
-     * @param outputFile The file to write to
-     * @param arch The arch of the shared object being analysed
-     * @return true if the file was written successfully, else false
-     * @return true if the file was written successfully, else false
+     * @param stdout The input stream
+     * @param outputFile The output file
      */
-    boolean outPutSymbolFile(BufferedReader outReader, File outputFile, String arch) {
-        // Output the file from stdout
-        try {
-            FileOutputStream is = new FileOutputStream(outputFile)
-            OutputStreamWriter osw = new OutputStreamWriter(is)
-            Writer writer = new BufferedWriter(osw)
-
-            Pattern addressPattern = Pattern.compile("^\\s+([0-9a-f]+):", Pattern.CASE_INSENSITIVE)
-            boolean justSeenAddress = false
-            String previousAddress = null
-
-            // Loop to remove redundant address lines (just keep the first and last addresses of each block)
-            String line = outReader.readLine()
-            Matcher addressMatcher
-            while (line != null) {
-
-                // Check to see if the current line is an address
-                addressMatcher = addressPattern.matcher(line)
-                if (addressMatcher.find()) {
-
-                    // Only output the line if this is the start of a block of addresses
-                    if (!justSeenAddress) {
-                        writer.writeLine(line)
-                        previousAddress = null
-                    } else {
-                        previousAddress = line
-                    }
-
-                    justSeenAddress = true
-                } else {
-
-                    // If this is the end of a block of addresses then output the last address
-                    if (justSeenAddress && previousAddress != null) {
-                        writer.writeLine(previousAddress)
-                    }
+    static void outputZipFile(InputStream stdout, File outputFile) {
+        GZIPOutputStream zipStream = new GZIPOutputStream(new FileOutputStream(outputFile));
 
-                    writer.writeLine(line)
-
-                    previousAddress = null
-                    justSeenAddress = false
-                }
-
-                line = outReader.readLine()
-            }
-
-            writer.close()
-        } catch (IOException e) {
-            project.logger.error("failed to write symbols for " + arch + ": " + e.getMessage())
-            return false
+        byte[] buffer = new byte[1024];
```

Comment:
BufferedOutputStream defaults to `8192`, we could probably increase this to speed up IO

Reply:
changed